    Multinomial Logistic Regression on Fashion-MNIST with sklearn
        by Carter Hempstead
    CSCI 4961, RPI Fall '21
    Original Github Repo: https://github.com/zalandoresearch/fashion-mnist
    Cited: arXiv:1708.07747

First, we need to load the Fashion-MNIST dataset into Python as instructed in the 'Usage' section of the 'README.md' file in the repo. Loading the data utilizes NumPy and the mnist_reader function supplied in the repo

In [30]:
import mnist_reader
X_train, y_train = mnist_reader.load_mnist('', kind='train')
X_test, y_test = mnist_reader.load_mnist('', kind='t10k')

Now X_train contains the images and y_train contains the corresponding labels for the model to use for training. Similarly, X_test and y_test contain the images and respective labels that the model will use to test on. 

In [45]:
print("Training Images:" , X_train.shape)
print("Training Labels:", y_train.shape)
print("Testing Images:" , X_test.shape)
print("Testing Labels:", y_test.shape)

Training Images: (60000, 784)
Training Labels: (60000,)
Testing Images: (10000, 784)
Testing Labels: (10000,)


We can print the training and testing data to see the amount of images and labels and the dimensions of the images. After printing we can see that we have 60,000 training labels and 60,000 corresponding training images of 28x28 pixel resolution(784 pixels). Also, for the test data we have 10,000 testing images of the same resolution, 28x28, and 10,000 corresponding testing labels. The training and testing labels are just a digit 0-9 representing each of the 10 different clothing items.